### PR TITLE
add support for Umami analytics

### LINKED ISSUE #327: feature request: support for Umami analytics
**Feature Description**

It would be nice if there was out-of-the-box support for [Umami analytics](https://umami.is/).

**Problem/Solution**

Umami is a self-hosted analytics solution with arguably much better privacy properties than Google Analytics.

**Alternatives Considered**

I could modify the template myself to include the required JavaScript. However, this seems like a useful feature to have for a larger number of users, and it would be nice if Umami support could be enabled by simply configuring it in the config.

### PR DESCRIPTION
**Pull request by marten-seemann**
Fixes #327.

### CODE REVIEW

**Comment by imfing** [comment]
Hi, thank you for the PR. 

I don't plan to include this in the next release, but will kep this open for the future.

**Comment by marten-seemann** [comment]
@imfing Any changes you’d like to see to this PR? Any chance this could be included in the next release? I’ve already tested the code and it works, would be great if I could remove my fork with the next release.

**Comment by imfing** [comment]
Just a side note: for current versions (`0.7.x`), the tracking code could be placed under `layouts/partials/custom/head-end.html` of your site

```html
<script async defer src="https://url" data-website-id="id" ></script>
```

File: layouts/partials/head.html [hunk 1]
```diff
@@ -36,6 +36,11 @@
     {{ partial "google-analytics.html" . }}
   {{- end }}
 
+  <!-- Umami -->
+  {{- if and hugo.IsProduction .Site.Params.umami.serverURL .Site.Params.umami.id }}
```
Comment (imfing) [inline layouts/partials/head.html:40]: I suggest placing it under `analytics`:

```yaml
params:
  analytics:
    # https://github.com/umami-software/umami
    umami:
      server: https://server-url
      # script: umami.js
      id: "site-id"
```

**Review by imfing — CHANGES_REQUESTED** [review]
Per above, also could you create another PR to add documentation for this?

**Pull request closed without merging**